pkgs/applications/networking/nextcloud-client-bin/generic.nix

```
@@ -0,0 +1,50 @@
+{ darwin
+, p7zip
+, libarchive
+, stdenv
+, lib
+, fetchurl
+}:
+{ pname, version, url, hash }:
+
+let
+  inherit (stdenv) targetPlatform;
+  ARCH = if targetPlatform.isArch64 then "arm64" else "x64";
+in stdenv.mkDerivation rec {
+  inherit pname version;
+
+  src = fetchurl {
+    inherit url hash;
+  };
+
+  unpackPhase = lib.optionalString stdenv.isDarwin
+  ''
+    7z x $src
+    bsdtar -xf Payload~
+  '';
+
+  nativeBuildInputs = lib.optionalString stdenv.isDarwin [
+    p7zip
+    libarchive
+  ];
+
+  installPhase = lib.optionalString stdenv.isDarwin
+  ''
+    runHook preInstall
+
+    mkdir -p $out/Applications
+    ls
+    cp -R Applications/Nextcloud.app $out/Applications
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    homepage = "https://nextcloud.com";
+    license = licenses.gpl2Plus;
+    maintainers = [  ];
+    platforms = platforms.darwin;
+    mainProgrm = "nextcloud";
+  };
+}
```

```suggestion
in stdenv.mkDerivation (finalAttrs: {
  inherit pname version;

  src = fetchurl {
    inherit url hash;
  };

  unpackPhase = lib.optionalString stdenv.isDarwin
  ''
    7z x $src
    bsdtar -xf Payload~
  '';

  nativeBuildInputs = lib.optionalString stdenv.isDarwin [
    p7zip
    libarchive
  ];

  installPhase = lib.optionalString stdenv.isDarwin
  ''
    runHook preInstall

    mkdir -p $out/Applications
    ls
    cp -R Applications/Nextcloud.app $out/Applications

    runHook postInstall
  '';

  meta = with lib; {
    homepage = "https://nextcloud.com";
    license = licenses.gpl2Plus;
    maintainers = [  ];
    platforms = platforms.darwin;
    mainProgrm = "nextcloud";
  };
})
```

https://discourse.nixos.org/t/avoid-rec-expresions-in-nixpkgs/8293

---

Is there a particular reason for this? I am asking for learning purposes. Is there a functional difference or is it there to keep with convention?

---

> Is there a particular reason for this? I am asking for learning purposes. Is there a functional difference or is it there to keep with convention?

drupol explained it pretty well here https://github.com/NixOS/nixpkgs/pull/299032#discussion_r1538792904

---

Changed to the recommendation